skrifa/src/color/mod.rs

```diff
@@ -1,6 +1,406 @@
+    RadialGradient {
+        c0: Point<f32>,
+        r0: f32,
+        c1: Point<f32>,
+        r1: f32,
+        color_stops: &'a [ColorStop],
+        extend: Extend,
+    },
+    /// A sweep gradient, also called conical gradient. The color stops are
+    /// normalized to the range from 0 to 1 and the returned angles are to be
+    /// interpreted in _clockwise_ direction (swapped from the meaning in the
+    /// font file).  The stop normalization may mean that the angles may be
+    /// larger or smaller than the range of 0 to 360. Note that only the range
+    /// from 0 to 360 degrees is to be drawn, see
+    /// <https://learn.microsoft.com/en-us/typography/opentype/spec/colr#sweep-gradients>.
+    SweepGradient {
+        c0: Point<f32>,
+        start_angle: f32,
+        end_angle: f32,
+        color_stops: &'a [ColorStop],
+        extend: Extend,
+    },
+}
+
+/// Result type used in [`draw_color_glyph`](ColorPainter::draw_color_glyph)
+/// through which the client can signal whether a COLRv1 glyph referenced by
+/// another COLRv1 glyph can be drawn from cache or whether the glyphs subgraph
+/// should be traversed.
+pub enum ColrGlyphDrawResult {
+    ColrGlyphDrawn,
+    ColrGlyphDrawingNotImpl,
+}
+
+/// A group of required callbacks to be provided by the client. Each callback is
+/// executing a particular drawing or canvas transformation operation. The
+/// trait's callback functions are invoked when [`paint`](ColorPaintable::paint) is
+/// called with a [`ColorPainter`] trait object. The documentation for each
+/// function describes what actions are to be executed using the client side 2D
+/// graphics API, usually by performing some kind of canvas operation.
+pub trait ColorPainter {
+    /// Push the specified transform by concatenating it to the current
+    /// transformation matrix.
+    fn push_transform(&mut self, transform: Transform);
+    /// Restore the transformation matrix to the state before the previous
+    /// [`push_transform`](ColorPainter::push_transform) call.
+    fn pop_transform(&mut self);
+
+    /// Apply a clip path in the shape of glyph specified by `glyph_id`.
+    fn push_clip_glyph(&mut self, glyph_id: GlyphId);
+    /// Apply a clip rectangle specified by `clip_rect`.
+    fn push_clip_box(&mut self, clip_box: BoundingBox<f32>);
+    /// Restore the clip state to the state before a previous
+    /// [`push_clip_glyph`](ColorPainter::push_clip_glyph) or
+    /// [`push_clip_box`](ColorPainter::push_clip_box) call.
+    fn pop_clip(&mut self);
+
+    /// Fill the current clip area with the specified gradient fill.
+    fn fill(&mut self, fill_type: FillType);
+
+    /// Optionally implement this method: Draw an unscaled COLRv1 glyph given
+    /// the current transformation matrix (as accumulated by
+    /// [`push_transform`](ColorPainter::push_transform) calls).
+    fn draw_color_glyph(&mut self, _glyph: GlyphId) -> Result<ColrGlyphDrawResult, ReadError> {
```

Comment:
As above, I think `paint_cached_color_glyph` better captures the intent. This should also have the same error type as `paint`.

Reply:
Renamed to `paint_cached_color_glyph`.